pkg/lib/qunit-tests.ts

```diff
@@ -87,4 +87,22 @@ qunitTap(QUnit, function(message: string, ...args: unknown[]) {
     console.log(message, args);
 });
 
+export function f(format: TemplateStringsArray, ...args: unknown[]) {
+    const strings = [...format].reverse();
+    args.reverse();
+
+    const parts = [strings.pop()];
+    if (strings.length !== args.length) {
+        throw new Error('unequal strings and args in f-string');
+    }
+
+    while (args.length !== 0) {
+        const arg = args.pop();
+        parts.push(JSON.stringify(arg) || String(arg));
```

Comment:
The only case where JSON.stringify() returns something false-y that I know of is calling it with `undefined`. All other "weird" values like `[undefined]`, `NaN`, `{'a': NaN}` etc. are converted to `[null]`, `null`, etc. Are you aware of any other case here?

(This isn't a negative review at all, just curiosity)

Reply:
I indeed did this for the undefined case (which appears often in tests).

I guess we might use this with other non-JSON things in the future so it seemed nice to have the fallback...

Reply:
Right, it is nice, I was just wondering if you found another curious JS behaviour :grin: